infra/modules/gke-cluster/variables.tf

```diff
@@ -43,10 +41,34 @@ variable "name_suffix" {
   EOT
 }
 
-variable "service_account" {
+variable "network" {
   type        = string
+  description = "Google Cloud VPC network in which the cluster will be created"
+}
+
+variable "google_service_account" {
+  type = object({
+    email = string
+    name  = string
```

Comment:
Nit: we use the word `name` here, but `id` above. Consider changing this to ID

Reply:
This is referencing the actual "google_service_account" resource from Terraform and we specifically reference the name and email fields of this object inside this module; so we have to use the same fields.